<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1184</width>
    <height>775</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>My Main Window</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="groupBox_calibration">
      <property name="title">
       <string>标定</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="0" column="0" rowspan="3" colspan="4">
        <widget class="QGroupBox" name="groupBox_RobotBase2LocationBase_calib">
         <property name="title">
          <string>机器人基座标与定位器基座标标定</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_3">
          <item row="0" column="0">
           <widget class="QPushButton" name="pushButton_mark_point1">
            <property name="text">
             <string>标定点一</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="2">
           <widget class="QPushButton" name="pushButton_mark_point2">
            <property name="text">
             <string>标定点二</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QPushButton" name="pushButton_mark_point3">
            <property name="text">
             <string>标定点三</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="pushButton_mark_point4">
            <property name="text">
             <string>标定点四</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1" colspan="2">
           <widget class="QPushButton" name="pushButton_mark_point5">
            <property name="text">
             <string>标定点五</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QPushButton" name="pushButton_mark_point6">
            <property name="text">
             <string>标定点六</string>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QPushButton" name="pushButton_delete_mark_result">
            <property name="text">
             <string>删除标定结果</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1" colspan="2">
           <widget class="QPushButton" name="pushButton_save_mark_result">
            <property name="text">
             <string>保存标定结果</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QPushButton" name="pushButton_compute">
            <property name="text">
             <string>计算</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1" colspan="2">
           <widget class="QLabel" name="label_linear_error_tag_2">
            <property name="text">
             <string>标定误差</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="4" column="2" colspan="2">
           <widget class="QCheckBox" name="checkBox_use_track2tcp">
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="text">
             <string>追踪器转置TCP点位</string>
            </property>
           </widget>
          </item>
          <item row="3" column="3">
           <widget class="QLabel" name="label_calibration_error">
            <property name="text">
             <string/>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="4" colspan="2">
        <widget class="QGroupBox" name="groupBox_tracker2tcp_calib">
         <property name="title">
          <string>Tracker &lt;-&gt; TCP 标定</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QPushButton" name="pushButton_tracker2tcp_markpoint">
            <property name="text">
             <string>记录点</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="2">
           <widget class="QLabel" name="label_tracker2tcp_marked_num">
            <property name="text">
             <string>需6个点,已记录点数: 0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="pushButton_tracker2tcp_clear_point">
            <property name="text">
             <string>清除所有点</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="pushButton_tracker2tcp_calibrate_ori">
            <property name="text">
             <string>标定姿态</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="pushButton_tracker2tcp_calibrate_pos">
            <property name="text">
             <string>标定位置</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="4" rowspan="2" colspan="2">
        <widget class="QGroupBox" name="groupBox_flang2tcp_calib">
         <property name="title">
          <string>法兰盘 &lt;-&gt; TCP 标定</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="0" column="0">
           <widget class="QPushButton" name="pushButton_flange2tcp_markpoint">
            <property name="text">
             <string>记录点</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="label_flange2tcp_marked_num">
            <property name="text">
             <string>需6个点,已记录点数: 0</string>
            </property>
            <property name="alignment">
             <set>Qt::AlignCenter</set>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="pushButton_flange2tcp_clear_point">
            <property name="text">
             <string>清除所有点</string>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QPushButton" name="pushButton_flange2tcp_calibrate">
            <property name="text">
             <string>标定</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_linear_error_tag">
         <property name="text">
          <string>线性误差</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="3" column="1" colspan="2">
        <widget class="QLabel" name="label_linear_error">
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item row="3" column="3">
        <widget class="QPushButton" name="pushButton_once_get">
         <property name="text">
          <string>获取</string>
         </property>
        </widget>
       </item>
       <item row="3" column="4">
        <widget class="QPushButton" name="pushButton_connect_ctr">
         <property name="text">
          <string>连接控制器</string>
         </property>
        </widget>
       </item>
       <item row="3" column="5">
        <widget class="QPushButton" name="pushButton_disconnect_ctr">
         <property name="text">
          <string>断开控制器</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_vive">
      <property name="title">
       <string>VIVE</string>
      </property>
      <widget class="QLabel" name="label_2">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>40</y>
         <width>61</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>实时坐标</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_vive_x_tag">
       <property name="geometry">
        <rect>
         <x>100</x>
         <y>20</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>x</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_vive_y_tag">
       <property name="geometry">
        <rect>
         <x>170</x>
         <y>20</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>y</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_vive_z_tag">
       <property name="geometry">
        <rect>
         <x>250</x>
         <y>20</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>z</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_vive_C_tag">
       <property name="geometry">
        <rect>
         <x>470</x>
         <y>20</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>C</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_vive_B_tag">
       <property name="geometry">
        <rect>
         <x>410</x>
         <y>20</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>B</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_vive_A_tag">
       <property name="geometry">
        <rect>
         <x>330</x>
         <y>20</y>
         <width>21</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string>A</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_vive_z">
       <property name="geometry">
        <rect>
         <x>230</x>
         <y>40</y>
         <width>61</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_vive_B">
       <property name="geometry">
        <rect>
         <x>380</x>
         <y>40</y>
         <width>61</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_vive_y">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>40</y>
         <width>61</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_vive_x">
       <property name="geometry">
        <rect>
         <x>90</x>
         <y>40</y>
         <width>61</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_vive_C">
       <property name="geometry">
        <rect>
         <x>450</x>
         <y>40</y>
         <width>61</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="label_vive_A">
       <property name="geometry">
        <rect>
         <x>310</x>
         <y>40</y>
         <width>61</width>
         <height>21</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_vive_start">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>80</y>
         <width>75</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>开始</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_vive_stop">
       <property name="geometry">
        <rect>
         <x>110</x>
         <y>80</y>
         <width>75</width>
         <height>23</height>
        </rect>
       </property>
       <property name="text">
        <string>停止</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_refresh_rate">
       <property name="geometry">
        <rect>
         <x>320</x>
         <y>80</y>
         <width>111</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
      <widget class="QLabel" name="label_refresh_rate_tag">
       <property name="geometry">
        <rect>
         <x>250</x>
         <y>80</y>
         <width>71</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>刷新率</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_refresh_rate">
       <property name="geometry">
        <rect>
         <x>440</x>
         <y>80</y>
         <width>61</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>设置</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_start_record">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>150</y>
         <width>91</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>开始录制</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_end_record">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>150</y>
         <width>91</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>结束录制</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_start_plackback">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>210</y>
         <width>91</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>开始回放</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_end_playback">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>210</y>
         <width>91</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>结束回放</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_parse_chart">
       <property name="geometry">
        <rect>
         <x>440</x>
         <y>140</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>选择绘图</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="checkBox_filtering">
       <property name="geometry">
        <rect>
         <x>610</x>
         <y>180</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>滤波</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_filter_window_size">
       <property name="geometry">
        <rect>
         <x>438</x>
         <y>180</y>
         <width>41</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>窗口</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_polynomial_order">
       <property name="geometry">
        <rect>
         <x>438</x>
         <y>220</y>
         <width>41</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>系数</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_filter_window_size">
       <property name="geometry">
        <rect>
         <x>520</x>
         <y>180</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_polynomial_order">
       <property name="geometry">
        <rect>
         <x>520</x>
         <y>220</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_traj_filtering">
       <property name="geometry">
        <rect>
         <x>570</x>
         <y>140</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>轨迹滤波</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="checkBox_label_interpolation_interval">
       <property name="geometry">
        <rect>
         <x>730</x>
         <y>220</y>
         <width>121</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>按时间步距插补</string>
       </property>
      </widget>
      <widget class="QLineEdit" name="lineEdit_label_interpolation_interval">
       <property name="geometry">
        <rect>
         <x>840</x>
         <y>180</y>
         <width>81</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QLabel" name="label_interpolation_interval">
       <property name="geometry">
        <rect>
         <x>730</x>
         <y>180</y>
         <width>91</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>时间步距 ms</string>
       </property>
      </widget>
      <widget class="QCheckBox" name="checkBox_end_with_chart">
       <property name="geometry">
        <rect>
         <x>270</x>
         <y>150</y>
         <width>121</width>
         <height>31</height>
        </rect>
       </property>
       <property name="text">
        <string>结束时绘图</string>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
